### PR TITLE
Remove useless J9_RUNTIME_DENY_ILLEGAL_ACCESS flag

### PR DESCRIPTION
**Pull request by keithc-ca**
* no need to set a flag nobody checks
* must still set system property "jdk.module.illegalAccess" for openjdk
* tidy up in `addPropertyForOptionWithEqualsArg()`

### CODE REVIEW

File: runtime/vm/vmprops.c [hunk 1]
```diff
@@ -494,8 +494,12 @@ addModularitySystemProperties(J9JavaVM * vm)
 		/* check if SYSPROP_JDK_MODULE_ILLEGALACCESS has been set */
 		J9VMSystemProperty *property = NULL;
 
-		if (J9SYSPROP_ERROR_NOT_FOUND != getSystemProperty(vm, SYSPROP_JDK_MODULE_ILLEGALACCESS, &property)) {
-
+		if (J9SYSPROP_ERROR_NOT_FOUND == getSystemProperty(vm, SYSPROP_JDK_MODULE_ILLEGALACCESS, &property)) {
+#if JAVA_SPEC_VERSION >= 17
+			/* beginning with Java 17, the default is --illegal-access=deny */
+			vm->runtimeFlags |= J9_RUNTIME_DENY_ILLEGAL_ACCESS;
```
Comment (JasonFengJ9) [inline runtime/vm/vmprops.c:500]: This is an unnecessary change for now.

`J9_RUNTIME_DENY_ILLEGAL_ACCESS` is not used which is being investigated by @ChengJin01 at https://github.com/eclipse/openj9/issues/11523

Comment (keithc-ca) [reply runtime/vm/vmprops.c:500]: That doesn't change the fact that the default is different in Java 17.
When `J9_RUNTIME_DENY_ILLEGAL_ACCESS` is used as appropriate, we will have the right behavior in all versions (including Java 17).

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
jenkins compile zlinux jdknext

**Comment by gacholio** [comment]
jenkins compile xlinux jdk8

**Comment by pshipton** [comment]
This should apply to jdk16+ as JEP 396 is part of the Java 16 plan.

http://openjdk.java.net/projects/jdk/16/

**Comment by pshipton** [comment]
Since this flag isn't used, I suggest it be deleted rather than updating it.

**Comment by gacholio** [comment]
I no longer understand if this PR is needed, so I'm not merging it.

**Comment by keithc-ca** [comment]
> Since this flag isn't used, I suggest it be deleted rather than updating it.

I would expect that the solution to #11523 will involve paying attention to that flag.

**Comment by keithc-ca** [comment]
Updated for Java 16+.

**Comment by pshipton** [comment]
> I would expect that the solution to #11523 will involve paying attention to that flag.

I doubt it, all the work is done by OpenJDK, the solution likely involves calling the right thing(s).

**Comment by keithc-ca** [comment]
Let's park this until #11523 is addressed at which point we should know whether the flag is needed. Then we can either merge this, or remove the flag.

**Comment by pshipton** [comment]
https://github.com/eclipse/openj9/issues/11523 doesn't need J9_RUNTIME_DENY_ILLEGAL_ACCESS, I suggest it be deleted.

**Comment by keithc-ca** [comment]
Rebased and repurposed to remove the unused flag.

**Comment by keithc-ca** [comment]
Jenkins compile xlinux jdk8

**Comment by keithc-ca** [comment]
Jenkins compile zlinux jdknext

File: runtime/oti/j9consts.h [hunk 2]
```diff
@@ -276,7 +276,6 @@ extern "C" {
 #define J9_RUNTIME_ARGENCODING_UNICODE 0x2000
 #define J9_RUNTIME_ARGENCODING_LATIN 0x4000
 #define J9_RUNTIME_ARGENCODING_UTF8 0x8000
-#define J9_RUNTIME_DENY_ILLEGAL_ACCESS 0x10000
```
Comment (gacholio) [inline runtime/oti/j9consts.h:279]: Please replace this with J9_RUNTIME_UNUSED_0x10000 instead of just removing the line.

Comment (keithc-ca) [reply runtime/oti/j9consts.h:279]: I started to do that, but there are four unused bits: would you like a macro like this for each?
```
#define J9_RUNTIME_UNUSED_0x10000 0x10000
#define J9_RUNTIME_UNUSED_0x20000 0x20000
#define J9_RUNTIME_UNUSED_0x40000 0x40000
#define J9_RUNTIME_UNUSED_0x80000 0x80000
```

Comment (gacholio) [reply runtime/oti/j9consts.h:279]: Please - I fixed one today that was just called FLAG1, which would be especially hard to maintain.

Comment (keithc-ca) [reply runtime/oti/j9consts.h:279]: Placeholder macros added.

**Comment by keithc-ca** [comment]
The addition of the placeholder macros should not invalidate these builds:
* https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDK8_x86-64_linux/2624/
* https://ci.eclipse.org/openj9/job/Pipeline_Build_Test_JDKnext_s390x_linux/794/

**Review by gacholio — APPROVED** [review]

**Comment by gacholio** [comment]
The above builds passed.
